### PR TITLE
Opt out of IMip CalDAV Plugin

### PR DESCRIPTION
**Pull request by georgehrke**
This pull-request makes it possible to opt out of the iMip Plugin for CalDAV. If disabled, the server will no longer send out emails to attendees. This was requested by many people.

I already implemented this earlier in https://github.com/nextcloud/server/pull/5304, but removed it again because of requested changes concerning the implementation.

In contrast to https://github.com/nextcloud/server/pull/5304, this pull-request doesn't use the `config.php`, but stores it in the database.

This pull-request adds a new GUI element:
c3285bf8-7c80-4ac1-938d-e4129d1a2af1

Config method besides GUI:
```
php occ config:app:set dav sendInvitations --value yes/no
```

### CODE REVIEW

**Comment by georgehrke** [comment]
to test this:
- setup email properly
- make sure option is enabled
- create event (in the future) with attendee
- make sure email was sent
- disable option 
- create event (in the future) with attendee
- make sure email was not sent

File: apps/dav/appinfo/v1/caldav.php [hunk 1]
```diff
@@ -52,6 +52,7 @@
 $calDavBackend = new CalDavBackend($db, $principalBackend, $userManager, $random, $dispatcher, true);
 
 $debugging = \OC::$server->getConfig()->getSystemValue('debug', false);
+$sendInvitations = (\OC::$server->getConfig()->getAppValue('dav', 'sendInvitations', 'yes') === 'yes');
```
Comment (nickvergessen) [inline apps/dav/appinfo/v1/caldav.php]: Remove brackets

File: apps/dav/js/settings-admin-caldav.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+"use strict";
+
+$('#caldavSendInvitations').change(function() {
+	var url = OC.generateUrl('/apps/dav/setSendInvitations');
```
Comment (nickvergessen) [inline apps/dav/js/settings-admin-caldav.js]: Use `OCP.AppConfig.setValue('dav', 'sendInvitations', ...)`

Then you can drop your controller as well

File: apps/dav/lib/Server.php [hunk 3]
```diff
@@ -76,6 +76,7 @@ public function __construct(IRequest $request, $baseUri) {
 		$mailer = \OC::$server->getMailer();
 		$dispatcher = \OC::$server->getEventDispatcher();
 		$timezone = new TimeFactory();
+		$sendInvitations = (\OC::$server->getConfig()->getAppValue('dav', 'sendInvitations', 'yes') === 'yes');
```
Comment (nickvergessen) [inline apps/dav/lib/Server.php]: remove brackets

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: apps/dav/js/settings-admin-caldav.js (hunk 2 above)
Comment (georgehrke) [reply apps/dav/js/settings-admin-caldav.js]: Good hint! Will change that accordingly

**Comment by georgehrke** [comment]
removed brackets and using `OCP.AppConfig` now ✅

**Review by nickvergessen — APPROVED** [review]

File: apps/dav/appinfo/v1/caldav.php [hunk 4]
```diff
@@ -52,6 +52,7 @@
 $calDavBackend = new CalDavBackend($db, $principalBackend, $userManager, $random, $dispatcher, true);
 
 $debugging = \OC::$server->getConfig()->getSystemValue('debug', false);
+$sendInvitations = \OC::$server->getConfig()->getAppValue('dav', 'sendInvitations', 'yes') === 'yes';
```
Comment (nickvergessen) [inline apps/dav/appinfo/v1/caldav.php]: Remove this useless variable and put the code directly into the if?

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by LukasReschke — APPROVED** [review]

File: apps/dav/appinfo/v1/caldav.php (hunk 4 above)
Comment (georgehrke) [reply apps/dav/appinfo/v1/caldav.php]: wouldn't `if(\OC::$server->getConfig()->getAppValue('dav', 'sendInvitations', 'yes') === 'yes') { ...` look really weird?

Comment (nickvergessen) [reply apps/dav/appinfo/v1/caldav.php]: well same goes for this assignment ;)

**Comment by kosli** [comment]
I tested the latest 13.0 release together with thunderbird lightning and even if the sending of the invitations is disabled on the NC side, TB doesn't ask about sending emails. Only if i disable the line with the schedule plugin ($server->addPlugin(new \OCA\DAV\CalDAV\Schedule\Plugin());) then it works as expected -> TB asks if it should send an invitation to the attendees. Can somebody confirm this? (after doing the change in Server.php you need to use the "reset calendar cache" option in TB).

**Pull request merged**